### ISSUE TITLE
Grab Emulator /Simulator

### ISSUE THREAD

**Issue by TehTurk** [Frooxius/NeosPublic#424]
This has been mentioned a couple times by me H3 and Lewis, basically having grabbers on objects that aren't your hands. The behavior could default to your main hand if you say want it controlled by a certain controller.

**Comment by H3BO3** [comment]
I would suggest keeping components like this more open-ended, obviously would need a user to control it but with appropriate logix nodes, so you would implement whatever controls you wanted. Natively basing it off controllers doesn't make much sense as that behavior could just be linked up with the right logix.

**Comment by GearBell** [comment]
leaving a comment here to support the need for this - itd be very useful for gadgets or avatar rigs; multiple armed avatars would be able to grab with the puppet hands, and gagets like extending claws could pick stuff up/grab a wall for you to climb. theres many many uses for this!

**Comment by ProbablePrime** [comment]
Can't Ryuvi pick stuff up with their feet? How is that done?

**Comment by mralext20** [comment]
i believe grox also has a 'pick up with feet'. AFAIK they move their grab anchor to their foot, and still use their controller to activate it.

**Comment by GearBell** [comment]
he moves his grab spheres. im currently doing the same for my extendo-arms, but it renders our real hands useless. with grab sphere emulators, its *additional*, not alternative

**Comment by GearBell** [comment]
its useful for gadgets and would make multi limbed avatars more immpressive if we had emulators, so that all hands can grab, even if they are simply copying what the master hands are doing

**Comment by Hayden-da-Serg** [comment]
I two want this, for a long while I've been wanting to be able to pick things up with my mouth but I also don't want to touch the normal grab spheres as I could see that causing a verity of issues down the line.

**Comment by TehTurk** [comment]
Well the main issue i've realized over time is if there's a way to designate a way to integrate into the Grabber on the CommonTool, while you could have emulation, it'd be easier if you could disable one and then write in the ref for another long as you only have 2 active at all times.

**Comment by GearBell** [comment]
but the whole point is to *not* disable any, but to *add*. hayden wants to additionally grab things with his mouth in addition to his hands, i think itd be neat to be able to hold stuff in all 4 arms, and coffee wanted it for gadgets, last i heard.

**Comment by H3BO3** [comment]
Yes me too, I'd love if I could make a tool with a grabsphere with an adjustable radius

**Comment by TehTurk** [comment]
No! I do too, I'm just trying to see how Froox would be implementing it technically.

**Comment by H3BO3** [comment]
Regarding implementation on this, I think it's pretty important that they could be able to function like ordinary grab spheres, but have open-ended options to make them as flexible as possible. Things like

- Radius
- Can climb climbable things (assuming that the user's locomotion module allows for it)
- Can pick up things

Just a few thoughts. I would really love to see these implemented, been waiting a while and I've recently gotten inspired more by a conversation with Linkbro...

**Comment by Psychpsyo** [comment]
This could also function in a similar way to TooltipSimulator, where you give a user and boolean to a component and it pretends to be a grabsphere.
